includes/api/class-wc-admin-rest-onboarding-plugins-controller.php

```diff
@@ -0,0 +1,229 @@
+
+		$api = plugins_api(
+			'plugin_information',
+			array(
+				'slug'   => sanitize_key( $slug ),
+				'fields' => array(
+					'sections' => false,
+				),
+			)
+		);
+
+		if ( is_wp_error( $api ) ) {
+			return new WP_Error( 'woocommerce_rest_plugin_install', __( 'The requested plugin could not be installed.', 'woocommerce-admin' ), 500 );
+		}
+
+		$upgrader = new Plugin_Upgrader( new Automatic_Upgrader_Skin() );
+		$result   = $upgrader->install( $api->download_link );
+
+		if ( is_wp_error( $result ) || is_null( $result ) ) {
+			return new WP_Error( 'woocommerce_rest_plugin_install', __( 'The requested plugin could not be installed.', 'woocommerce-admin' ), 500 );
+		}
+
+		return array(
+			'slug'   => $slug,
+			'name'   => $api->name,
+			'status' => 'success',
+		);
+	}
+
+	/**
+	 * Activate the requested plugin.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return array Plugin Status
+	 */
+	public function activate_plugin( $request ) {
+		$allowed_plugins = $this->get_allowed_plugins();
+		if ( ! in_array( $request['plugin'], array_keys( $allowed_plugins ), true ) ) {
+			return new WP_Error( 'woocommerce_rest_invalid_plugin', __( 'Invalid plugin.', 'woocommerce-admin' ), 404 );
+		}
+
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		$slug              = $request['plugin'];
+		$path              = $allowed_plugins[ $slug ];
+		$installed_plugins = get_plugins();
+
+		if ( ! in_array( $path, array_keys( $installed_plugins ), tre ) ) {
+			return new WP_Error( 'woocommerce_rest_invalid_plugin', __( 'Invalid plugin.', 'woocommerce-admin' ), 404 );
+		}
+
+		$result = activate_plugin( $path );
+		if ( ! is_null( $result ) ) {
+			return new WP_Error( 'woocommerce_rest_invalid_plugin', __( 'The requested plugin could not be activated.', 'woocommerce-admin' ), 500 );
+		}
+
+		return( array(
+			'slug'   => $slug,
+			'name'   => $installed_plugins[ $path ]['Name'],
+			'status' => 'success',
+		) );
+	}
+
+	/**
+	 * Get the schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'onboarding_plugin',
+			'type'       => 'object',
+			'properties' => array(
+				'plugins' => array(
+					'type'        => 'array',
+					'description' => __( 'Array of plugins that have been updated by the endpoint.', 'woocommerce-admin' ),
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+					'items'       => array(
+						'type'       => 'object',
+						'properties' => array(
+							'slug'   => array(
+								'description' => __( 'Plugin slug.', 'woocommerce-admin' ),
+								'type'        => 'string',
+								'context'     => array( 'view', 'edit' ),
+								'readonly'    => true,
+							),
+							'name'   => array(
+								'description' => __( 'Plugin dname.', 'woocommerce-admin' ),
```

Comment:
Accidental dcharacter here.

```suggestion
								'description' => __( 'Plugin name.', 'woocommerce-admin' ),
```